Warnings with newer clang and g++

Compiling a cmdstan model with clang 10 (I also believe the same thing happens for g++ 8) these warnings occurs:
```
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/DenseCoeffsBase.h: In instantiation of ‘class Eigen::DenseCoeffsBase<Eigen::ArrayWrapper<const Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double, double>, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> >, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> > > >, 0>’:
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/DenseBase.h:41:34:   required from ‘class Eigen::DenseBase<Eigen::ArrayWrapper<const Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double, double>, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> >, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> > > > >’
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/ArrayBase.h:39:34:   required from ‘class Eigen::ArrayBase<Eigen::ArrayWrapper<const Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double, double>, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> >, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> > > > >’
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/ArrayWrapper.h:42:7:   required from ‘class Eigen::ArrayWrapper<const Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double, double>, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> >, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> > > >’
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/BooleanRedux.h:144:30:   required from ‘bool Eigen::DenseBase<Derived>::hasNaN() const [with Derived = Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double, double>, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> >, const Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> > >]’
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/BooleanRedux.h:158:10:   required from ‘bool Eigen::DenseBase<Derived>::allFinite() const [with Derived = Eigen::CwiseUnaryOp<stan::math::value_of(EigMat&&) [with EigMat = const Eigen::Matrix<stan::math::var, -1, 1>&; stan::require_eigen_t<T>* <anonymous> = 0; stan::require_not_st_arithmetic<T>* <anonymous> = 0]::<lambda(auto:12&)> [with auto:12 = const Eigen::Matrix<stan::math::var, -1, 1>]::<lambda(const auto:13&)>, const Eigen::Matrix<stan::math::var, -1, 1> >]’
stan/lib/stan_math/stan/math/prim/err/check_finite.hpp:93:7:   required from ‘void stan::math::check_finite(const char*, const char*, const EigMat&) [with EigMat = Eigen::Matrix<stan::math::var, -1, 1>; stan::require_eigen_t<T>* <anonymous> = 0]’
stan/lib/stan_math/stan/math/prim/functor/ode_rk45.hpp:75:15:   required from ‘std::vector<Eigen::Matrix<typename stan::return_type<T_y0, T_t0, T_ts, Args ...>::type, -1, 1> > stan::math::ode_rk45_tol_impl(const char*, const F&, const Eigen::Matrix<RhsScalar, -1, 1>&, T_t0, const std::vector<T_l>&, double, double, long int, std::ostream*, const Args& ...) [with F = ode_model_namespace::dz_dt_odefunctor__; T_y0 = stan::math::var; T_t0 = int; T_ts = double; Args = {stan::math::var, stan::math::var, stan::math::var, stan::math::var}; typename stan::return_type<T_y0, T_t0, T_ts, Args ...>::type = stan::math::var; std::ostream = std::basic_ostream<char>]’
stan/lib/stan_math/stan/math/prim/functor/ode_rk45.hpp:188:27:   required from ‘std::vector<Eigen::Matrix<typename stan::return_type<T_y0, T_t0, T_ts, Args ...>::type, -1, 1> > stan::math::ode_rk45_tol(const F&, const Eigen::Matrix<RhsScalar, -1, 1>&, T_t0, const std::vector<T_l>&, double, double, long int, std::ostream*, const Args& ...) [with F = ode_model_namespace::dz_dt_odefunctor__; T_y0 = stan::math::var; T_t0 = int; T_ts = double; Args = {stan::math::var, stan::math::var, stan::math::var, stan::math::var}; typename stan::return_type<T_y0, T_t0, T_ts, Args ...>::type = stan::math::var; std::ostream = std::basic_ostream<char>]’
examples/ode/ode.hpp:428:21:   required from ‘T__ ode_model_namespace::ode_model::log_prob(std::vector<T_l>&, std::vector<int>&, std::ostream*) const [with bool propto__ = false; bool jacobian__ = false; T__ = stan::math::var; std::ostream = std::basic_ostream<char>]’
stan/src/stan/model/model_base_crtp.hpp:153:29:   required from ‘stan::math::var stan::model::model_base_crtp<M>::log_prob(std::vector<stan::math::var>&, std::vector<int>&, std::ostream*) const [with M = ode_model_namespace::ode_model; std::ostream = std::basic_ostream<char>]’
stan/src/stan/model/model_base_crtp.hpp:149:20:   required from here
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/DenseCoeffsBase.h:55:30: warning: ignoring attributes on template argument ‘Eigen::internal::packet_traits<double>::type’ {aka ‘__vector(2) doubl’} [-Wignored-attributes]
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/DenseCoeffsBase.h: In instantiation of ‘class Eigen::DenseCoeffsBase<Eigen::CwiseUnaryView<Eigen::MatrixBase<Eigen::Matrix<stan::math::var, -1, 1> >::val_Op, Eigen::Matrix<stan::math::var, -1, 1> >, 0>’:
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/DenseCoeffsBase.h:300:7:   required from ‘class Eigen::DenseCoeffsBase<Eigen::CwiseUnaryView<Eigen::MatrixBase<Eigen::Matrix<stan::math::var, -1, 1> >::val_Op, Eigen::Matrix<stan::math::var, -1, 1> >, 1>’
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/DenseCoeffsBase.h:551:7:   required from ‘class Eigen::DenseCoeffsBase<Eigen::CwiseUnaryView<Eigen::MatrixBase<Eigen::Matrix<stan::math::var, -1, 1> >::val_Op, Eigen::Matrix<stan::math::var, -1, 1> >, 3>’
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/DenseBase.h:41:34:   required from ‘class Eigen::DenseBase<Eigen::CwiseUnaryView<Eigen::MatrixBase<Eigen::Matrix<stan::math::var, -1, 1> >::val_Op, Eigen::Matrix<stan::math::var, -1, 1> > >’
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/MatrixBase.h:48:34:   required from ‘class Eigen::MatrixBase<Eigen::CwiseUnaryView<Eigen::MatrixBase<Eigen::Matrix<stan::math::var, -1, 1> >::val_Op, Eigen::Matrix<stan::math::var, -1, 1> > >’
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/CwiseUnaryView.h:101:7:   required from ‘class Eigen::CwiseUnaryViewImpl<Eigen::MatrixBase<Eigen::Matrix<stan::math::var, -1, 1> >::val_Op, Eigen::Matrix<stan::math::var, -1, 1>, Eigen::Dense>’
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/CwiseUnaryView.h:58:7:   [ skipping 4 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
stan/lib/stan_math/stan/math/rev/functor/cvodes_integrator.hpp:288:57:   required from ‘std::vector<Eigen::Matrix<typename stan::return_type<T_y0, T_t0, T_ts, T_Args ...>::type, -1, 1> > stan::math::cvodes_integrator<Lmm, F, T_y0, T_t0, T_ts, T_Args>::operator()() [with int Lmm = 2; F = ode_model_namespace::dz_dt_odefunctor__; T_y0 = double; T_t0 = int; T_ts = double; T_Args = {double, double, double, double}; typename stan::return_type<T_y0, T_t0, T_ts, T_Args ...>::type = double]’
stan/lib/stan_math/stan/math/rev/functor/ode_bdf.hpp:61:20:   required from ‘std::vector<Eigen::Matrix<typename stan::return_type<T_y0, T_t0, T_ts, Args ...>::type, -1, 1> > stan::math::ode_bdf_tol_impl(const char*, const F&, const Eigen::Matrix<RhsScalar, -1, 1>&, const T_t0&, const std::vector<T_l>&, double, double, long int, std::ostream*, const T_Args& ...) [with F = ode_model_namespace::dz_dt_odefunctor__; T_y0 = double; T_t0 = int; T_ts = double; T_Args = {double, double, double, double}; typename stan::return_type<T_y0, T_t0, T_ts, Args ...>::type = double; std::ostream = std::basic_ostream<char>]’
stan/lib/stan_math/stan/math/rev/functor/ode_bdf.hpp:106:26:   required from ‘std::vector<Eigen::Matrix<typename stan::return_type<T_y0, T_t0, T_ts, Args ...>::type, -1, 1> > stan::math::ode_bdf_tol(const F&, const Eigen::Matrix<RhsScalar, -1, 1>&, const T_t0&, const std::vector<T_l>&, double, double, long int, std::ostream*, const T_Args& ...) [with F = ode_model_namespace::dz_dt_odefunctor__; T_y0 = double; T_t0 = int; T_ts = double; T_Args = {double, double, double, double}; typename stan::return_type<T_y0, T_t0, T_ts, Args ...>::type = double; std::ostream = std::basic_ostream<char>]’
examples/ode/ode.hpp:420:20:   required from ‘T__ ode_model_namespace::ode_model::log_prob(std::vector<T_l>&, std::vector<int>&, std::ostream*) const [with bool propto__ = false; bool jacobian__ = false; T__ = double; std::ostream = std::basic_ostream<char>]’
stan/src/stan/model/model_base_crtp.hpp:147:29:   required from ‘double stan::model::model_base_crtp<M>::log_prob(std::vector<double>&, std::vector<int>&, std::ostream*) const [with M = ode_model_namespace::ode_model; std::ostream = std::basic_ostream<char>]’
stan/src/stan/model/model_base_crtp.hpp:144:17:   required from here
stan/lib/stan_math/lib/eigen_3.3.7/Eigen/src/Core/DenseCoeffsBase.h:55:30: warning: ignoring attributes on template argument ‘Eigen::internal::packet_traits<double>::type’ {aka ‘__vector(2) doubl’} [-Wignored-attributes]
```

I believe there is nothing we can do but set the `-Wno-ignored-attributes` flag
#### Current Version:
v3.2.0

--- a/make/compiler_flags
+++ b/make/compiler_flags
@@ -117,7 +117,7 @@ CXXFLAGS_SUNDIALS ?= -pipe
 # Update compiler flags with operating system specific modifications
 ##
 ifeq ($(OS),Windows_NT)
-  CXXFLAGS_WARNINGS ?= -Wall -Wno-unused-function -Wno-uninitialized -Wno-unused-but-set-variable -Wno-unused-variable -Wno-sign-compare -Wno-unused-local-typedefs
+  CXXFLAGS_WARNINGS ?= -Wall -Wno-unused-function -Wno-uninitialized -Wno-unused-but-set-variable -Wno-unused-variable -Wno-sign-compare -Wno-unused-local-typedefs -Wno-int-in-bool-context -Wno-attributes
   CPPFLAGS_GTEST ?= -DGTEST_HAS_PTHREAD=0
   CPPFLAGS_OS ?= -D_USE_MATH_DEFINES
   ## Detecting Process Bitness:
@@ -162,9 +162,8 @@ endif
 ## makes reentrant version lgamma_r available from cmath
 CXXFLAGS_OS += -D_REENTRANT
 
-ifeq (gcc,$(CXX_TYPE))
-CXXFLAGS_WARNINGS += -Wno-int-in-bool-context -Wno-attributes
-endif
+## silence warnings occuring due to the TBB and Eigen libraries
+CXXFLAGS_WARNINGS += -Wno-ignored-attributes
 
 ################################################################################
 # Setup OpenCL

